### PR TITLE
Add workflow to check PRs (build/test/check_fmt)

### PR DESCRIPTION
**Pull request by 0xkubectl**
Add a simple workflow to cross check if PRs and commits to main work.
Based on this: https://github.com/C0D3-M4513R/DexProtectOscRS/blob/main/.github/workflows/rust.yml

Depends on https://github.com/vrchatapi/vrchatapi-rust/pull/20

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "1.79.0"
```
Comment (C0D3-M4513R) [inline rust-toolchain.toml]: Explicitly setting the version might be a bad idea.

File: .github/workflows/check_pr.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  check-fmt:
+    runs-on: ubuntu-latest
+    name: Check formatting
+    steps:
+      - uses: actions/checkout@v4
+      - uses: moonrepo/setup-rust@v1
+        with:
+          components: rustfmt
+      - run: cargo fmt --all -- --check
```
Comment (C0D3-M4513R) [inline .github/workflows/check_pr.yaml:36]: I don't see a formatting job as useful.
Especially not, if the library itself isn't formatted properly anyways.
ref. #20

Comment (0xkubectl) [reply .github/workflows/check_pr.yaml:36]: This largely builds on [this change](https://github.com/vrchatapi/vrchatapi-rust/blob/978963b2bbfa60bc51f664b16d1c7d4613ab3829/generate.sh#L47) in this PR, kinda driveby i know - if need be ill put in a seperate PR.

Comment (C0D3-M4513R) [reply .github/workflows/check_pr.yaml:36]: I already split up two things from your pr.
Try to keep pr's to one thing.

File: rust-toolchain.toml (hunk 1 above)
Comment (0xkubectl) [reply rust-toolchain.toml]: In what way exactly? IMHO it makes sure all people working on the project use the same version.
Also its read by the setup-rust action.

Comment (C0D3-M4513R) [reply rust-toolchain.toml]: Yes, but there might be an openapi bump in the future, which might need a newer stable.
Just using the latest stable in ci ensures that the ci reflects what the publish result would be, if it was a release.

File: .github/workflows/check_pr.yaml (hunk 2 above)
Comment (C0D3-M4513R) [reply .github/workflows/check_pr.yaml:36]: You might want to remove those changes from your pr though.
Making smaller pr's is generally better.

Comment (0xkubectl) [reply .github/workflows/check_pr.yaml:36]: Sure thing! changed [here](https://github.com/vrchatapi/vrchatapi-rust/pull/17/commits/9be568c4fddd3139a0c6ce1a78befa7717ef7d81)

File: rust-toolchain.toml (hunk 1 above)
Comment (0xkubectl) [reply rust-toolchain.toml]: The generator is pinned aswell so I assume that what you describe could not happen. If the decision is made to update, one would have to bump the generator version and this.
Working with a lot of docker images and being burned by the `latest` tag in production, IMHO repoducible builds and consistent versioning outweight the conveniece of just getting the latest version.

Comment (0xkubectl) [reply rust-toolchain.toml]: Actually to be fair, the generate script is not idempotent as it does not pull a specific version of the spec. Which is another issue in of itself as this means one commit which could build at one point time, could fail at some other time.

To elaborate on why I think this is important (arguably contrived as we do not benchmark, work on such a low level that compiler decisions matter a whole lot or distribute compiled binaries, but for the sake of argument lets assume they do): Someone reports an issue with a specific version. Someone then tries to repoduce the issue with a different rust version and cannot do it, cuz the latest version uses a new LLVM version that sidesteps the root cause of the issue cuz it reorders some execution slightlty different. This makes confirming the issue and eventually debugging it becomes harder as you now have to factor in a range of versions the libary could have been compiled against. Assuming everything else is idempotent, when using a version that is tied to a commit allows for a ci build to repeatable and prove that a given commit works against a specific version of rust.

Comment (C0D3-M4513R) [reply rust-toolchain.toml]: I just don't think that reproducibility is what matters here.
I think that having a ci that just succeeds without any changes is more important.

The codegen being repeatable is also not an issue imo.
If you want the same codegen, you can always grab a specific version of the openapi-spec.
(I did that for #19 for example, to get rid of any extra changes)

Also anyone can compile their binary with any rust version they want. So we are going to have that hypothetical issue with LLVM regardless. As a library, we cannot make 100% repeatable builds. NixOs imo does a pretty good job of doing reproducible builds for rust binaries, but that's the closest you can come to bit-to-bit reproducible builds (and even then builds are sometimes not bit-to-bit reproducible).

Comment (0xkubectl) [reply rust-toolchain.toml]: I did some digging and the toolchain file is mostly used in projects that rely on nightly/unstable features to get some semblance of stability. As none are used, this is not as important. Ill remove it and update the action.

**Review by C0D3-M4513R — CHANGES_REQUESTED** [review]
Mixes #17 #20 and #19 in now.
I thought this pr was supposed to be a ci only change?

**Comment by 0xkubectl** [comment]
> Mixes #17 #20 and #19 in now. I thought this pr was supposed to be a ci only change?

The branch was forked of #17, didnt notice. Fixed.

**Review by C0D3-M4513R — APPROVED** [review]
This seems fine enough for now.
One thing, that doesn't really sit right with me is the on push trigger though, since that'll cause duplicate builds when an update is made to the specification.

**Pull request merged**